WordPress/Classes/Services/MediaThumbnailCoordinator.swift

```diff
@@ -25,13 +25,30 @@ class MediaThumbnailCoordinator: NSObject {
         return mediaThumbnailService
     }()
 
+    private lazy var mediaService: MediaService = {
+        let mediaService = MediaService(managedObjectContext: backgroundContext)
+        return mediaService
+    }()
+
     /// Tries to generate a thumbnail for the specified media object with the size requested
     ///
     /// - Parameters:
     ///   - media: the media object to generate the thumbail representation
     ///   - size: the size of the thumbnail
     ///   - onCompletion: a block that is invoked when the thumbnail generatio is completed with success or failure.
     @objc func thumbnail(for media: Media, with size: CGSize, onCompletion: @escaping ThumbnailBlock) {
+        if media.remoteStatus == .stub {
+            guard let mediaID = media.mediaID else {
+                onCompletion(nil, MediaThumbnailExporter.ThumbnailExportError.failedToGenerateThumbnailFileURL)
+                return
+            }
+            mediaService.getMediaWithID(mediaID, in: media.blog, withSuccess: { (media) in
+                self.thumbnail(for: media, with: size, onCompletion: onCompletion)
+            }) { (error) in
```

Comment:
I'd keep the failure parameter name here, as it can make things a little clearer to read.